Add --custom-function parameter to check.

umutphp opened the pull request:
In one of the projects that I have been working for a while, there are some legacy dump functions created and used frequently by the team. A sample function is;

`
function pre_dump( $var ) {
	echo '<pre>';
	var_dump( $var );
	echo '</pre>';
}
`

The check can not be able to find the usage of this kind of functions. With this pull request, user can give the names of such kind of custom debug functions to the check function with "--custom-function" parameter.

Commit:
Add --custom-function parameter to add custom functions to the list of functions to be checked

Commit:
Add test for custom function parameter and add .idea folder to gitignore

umutphp commented:
This will fix the issue #33.

JakubOnderka commented on .gitignore:
```diff
@@ -1,2 +1,3 @@
-/composer.lock
\ No newline at end of file
+/composer.lock
+.idea
```
Can you please revert that change? `.idea` folder is IDE specific and you should ignore it thought [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

umutphp replied:
Reverted the change.

JakubOnderka commented on README.md:
```diff
@@ -48,6 +48,7 @@ Options for run
 - `--doctrine` - check dump: `Doctrine::dump`, `\Doctrine\Common\Util\Debug::dump`
 - `--symfony` - check dump: `dump`, `VarDumper::dump`, `VarDumper::setHandler`
 - `--laravel` - check dump: `dd`, `dump`
+- `--custom-function` - comma separated custom function name(s) to check like "pre_echo"
```
Please use \` instead of `"` to match other parameters description

umutphp replied:
" is replaced with ` .

JakubOnderka commented on src/Settings.php:
```diff
@@ -131,6 +136,13 @@ public static function parseArguments(array $arguments)
         $setting->functionsToCheck = array_unique($setting->functionsToCheck);
+
+        // Any custom function is given
+        if (count($customFunctions) > 0) {
```
Better to use `empty()`

umutphp replied:
empty is used instead of count($customFunctions) > 0

umutphp committed:
Revert the .idea folder

JakubOnderka commented:
Thanks for your pull request, I am happy to add that functionality after you resolve conflict and requested changes. Thanks!

umutphp committed:
Implement the CR "Please use ` instead of " to match other parameters description"

umutphp committed:
Empty is used instead of count($customFunctions) > 0

umutphp committed:
Merge branch 'master' into master

kl3sk commented:
This is mergable ?

umutphp commented:
> Is this mergable ?

@kl3sk Technically yes, but I think the owner of the repo is a bit busy for a while.

kl3sk commented:
Ok, I didn't know it. Wait for it.

grogy committed:
Changed package name - new organization

grogy committed:
Changed package name - new organization

grogy committed:
Changed suggested and required-dev packages to new organization

grogy committed:
Changed Travis-CI icon

grogy committed:
Changed icon for download per month for new organization

grogy committed:
Changed last version icon by new organization

grogy committed:
Fixed running tests in PHP 5.4 and PHP 5.5 in CI

grogy committed:
Fixed Travis CI configuration

grogy committed:
Fixed Travis CI configuration

grogy committed:
Added replace to composer for previous package

grogy committed:
Added replace to composer for previous package

grogy committed:
Added changelog

grogy committed:
Added run CI with PHP 7.4

grogy committed:
Added run CI with PHP 7.4

grogy committed:
Fixed vendor names in readme for install

grogy committed:
Fixed vendor names in readme for install

grogy committed:
Created version 0.4

grogy committed:
Created version 0.4

reedy committed:
Create .gitattributes

reedy committed:
Create .gitattributes

grogy committed:
Added .gitattributes to changelog

grogy committed:
Added .gitattributes to changelog

peter279k committed:
Enhance test stuffs

peter279k committed:
Replace array syntax with short array syntax

peter279k committed:
Replace array syntax with short array syntax

grogy committed:
Added PHP array syntax to changelog

grogy committed:
Added PHP array syntax to changelog

grogy committed:
Added cleaning changes to changelog

grogy committed:
Added cleaning changes to changelog

jrfnl committed:
PHP 7.4: fix compatibility issue

Array access using curly braces has been deprecated in PHP 7.4.

Ref: https://www.php.net/manual/en/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace

jrfnl committed:
PHP 7.4: fix compatibility issue

Array access using curly braces has been deprecated in PHP 7.4.

Ref: https://www.php.net/manual/en/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace

jrfnl committed:
Composer: allow installation of more recent Parallel Lint

Using the caret is the recommended manner of setting a version requirement.

jrfnl committed:
Composer: allow installation of more recent Parallel Lint

Using the caret is the recommended manner of setting a version requirement.

grogy committed:
Added internal change to changelog

grogy committed:
Added internal change to changelog

grogy committed:
Added fix for PHP 7.4 to changelog

grogy committed:
Added fix for PHP 7.4 to changelog

umutphp committed:
Merge remote-tracking branch 'upstream/master'

umutphp committed:
Add test for none function founded

Caused by the review message in the following link.
https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470564342

umutphp committed:
Add test for none function founded

Caused by the review message in the following link.
https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470564342

umutphp committed:
Remove horizontal align

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470566935

umutphp committed:
Remove horizontal align

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470566935

umutphp committed:
Remove unneccessary empty check

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470562392

umutphp committed:
Remove unneccessary empty check

https://github.com/php-parallel-lint/PHP-Var-Dump-Check/pull/6#discussion_r470562392

umutphp committed:
Add the Symfony Vardumper dd() helper method introduced in 4.1
